lib/compiler-llvm/src/translator/code.rs

```diff
@@ -1048,6 +1048,7 @@ impl<'ctx, 'a> LLVMFunctionCodeGenerator<'ctx, 'a> {
         Ok(())
     }
 
+    #[allow(clippy::unnecessary_cast)]
```

Comment:
Why? 

Remove the cast instead?

Reply:
Same goes for all the others in this file.

Reply:
@theduke If I remove the cast, there might be unexpected issues on 32-bit vs 64-bit platforms, that's why I ignored the lint instead.

Reply:
I've removed it now with the lints that were working.